addons/sale_crm/models/sale_order.py

```diff
@@ -13,3 +13,23 @@ class SaleOrder(models.Model):
 
     def action_confirm(self):
         return super(SaleOrder, self.with_context({k:v for k,v in self._context.items() if k != 'default_tag_ids'})).action_confirm()
+
+    def _message_post_after_hook(self, new_message, message_values):
+        self._set_expected_revenue()
```

Comment:
Here, we only want to do something when we mark the so as sent. So I would add this condition above `self._set_expected_revenue()` : 

`if self.env.context.get('mark_so_as_sent'):`

Reply:
yes, you are right otherwise `update_revenues_from_so` will be called every time while posting a certain message.